internal/github/checkPendingCI.go

```diff
@@ -0,0 +1,127 @@
+package github
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/google/go-github/v60/github" // Ensure your go-github library version matches
+
+	"github.com/chia-network/github-bot/internal/config"
+)
+
+// CheckForPendingCI returns a list of PR URLs that are ready for CI to run but haven't started yet.
+func CheckForPendingCI(githubClient *github.Client, internalTeam string, cfg config.CheckStalePending) ([]string, error) {
+	teamMembers, _ := GetTeamMemberList(githubClient, internalTeam)
+	var pendingPRs []string
+
+	for _, fullRepo := range cfg.CheckStalePending {
+		log.Println("Checking repository:", fullRepo.Name)
+		parts := strings.Split(fullRepo.Name, "/")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("invalid repository name - must contain owner and repository: %s", fullRepo.Name)
+		}
+		owner, repo := parts[0], parts[1]
+
+		// Fetch community PRs using the FindCommunityPRs function
+		communityPRs, err := FindCommunityPRs(cfg.CheckStalePending, teamMembers, githubClient)
+		if err != nil {
+			return nil, err
+		}
+
+		for _, pr := range communityPRs {
+			// Dynamic cutoff time based on the last commit to the PR
+			lastCommitTime, err := getLastCommitTime(githubClient, owner, repo, pr.GetNumber())
+			if err != nil {
+				log.Printf("Error retrieving last commit time for PR #%d: %v", pr.GetNumber(), err)
+				continue
+			}
+			cutoffTime := lastCommitTime.Add(2 * time.Hour) // 2 hours after the last commit
+
+			if time.Now().Before(cutoffTime) {
+				log.Printf("Skipping PR #%d as it's still within the 2-hour window from the last commit.", pr.GetNumber())
+				continue
+			}
+
+			hasCIRuns, err := checkCIStatus(githubClient, owner, repo, pr.GetNumber())
+			if err != nil {
+				log.Printf("Error checking CI status for PR #%d: %v", pr.GetNumber(), err)
+				continue
+			}
+
+			teamMemberActivity, err := checkTeamMemberActivity(githubClient, owner, repo, pr.GetNumber(), teamMembers, lastCommitTime)
+			if err != nil {
+				log.Printf("Error checking team member activity for PR #%d: %v", pr.GetNumber(), err)
```

Comment:
This logline should contain the repo owner/name for more context, since this bot will be working with multiple repos.

Reply:
added owner/repo